Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,245 @@
+                            TextureData.WriteToFile(savePath, texture.Texture.EncodeToPNG());
+                        }
+
+                        // Setting the texture can fail if the asset database has not imported it yet.
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTextureSafe(textureName, null);
+                        }
+                        
+                        renderTarget.Release();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the texture on the material if it exists in the AssetDatabase, otherwise it schedules a delayed call to try again when the asset might be imported.
+        /// </summary>
+        /// <param name="mat">Material to set the texture on</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+            Texture2D savedTexture;
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+                if (savedTexture == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (UnityException) {
+                return false;
+            }
+
+            mat.SetTexture(textureName, savedTexture);
+
+            if (textureName == MeshSyncConstants._MetallicGlossMap ||
+                textureName == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma = true;
+                }
```

Comment:
Why would we need to call those? The asset importer marks the assets to be reimported if it detects it's necessary.

Reply:
> if it detects it's necessary.

If the asset is changed, then likely yes. But we are modifying the property of the importer here, and I am not sure if it can always detect that the importer settings are dirty.

There's also [SaveAndReimport()](https://docs.unity3d.com/ScriptReference/AssetImporter.SaveAndReimport.html) API which seems to support the idea that the Editor can't always detect dirty settings.

Reply:
Ok I've added those.